diff --git a/libass/ass_compat.h b/libass/ass_compat.h
index 38df913..47433d8 100644
--- a/libass/ass_compat.h
+++ b/libass/ass_compat.h
@@ -24,6 +24,10 @@
 #define _CRT_SECURE_NO_WARNINGS
 #define _USE_MATH_DEFINES
 #define inline __inline
+#include <crtversion.h>
+#if _VC_CRT_MAJOR_VERSION < 14
+#define snprintf _snprintf
+#endif
 #endif
 
 // Work around build failures on Windows with static FriBidi.
